src/SuperVault.sol

```diff
@@ -110,67 +149,75 @@ contract SuperVault is BaseStrategy, ISuperVault {
             /// @dev this superVault only supports superforms that have the same asset as the vault
             (superform,,) = superformIds_[i].getSuperform();
 
-            if (!factory.isSuperform(superformIds_[i])) {
+            if (!superformFactory.isSuperform(superformIds_[i])) {
                 revert SUPERFORM_DOES_NOT_EXIST(superformIds_[i]);
             }
 
             if (IBaseForm(superform).getVaultAsset() != asset_) {
                 revert SUPERFORM_DOES_NOT_SUPPORT_ASSET();
             }
 
+            /// @dev initial whitelist of superform IDs
+            _addToWhitelist(superformIds_[i]);
+
             totalWeight += startingWeights_[i];
         }
 
         if (totalWeight != TOTAL_WEIGHT) revert INVALID_WEIGHTS();
 
-        SV.numberOfSuperforms = numberOfSuperforms;
-        SV.superformIds = superformIds_;
-        SV.weights = startingWeights_;
-        SV.depositLimit = depositLimit_;
+        strategist = strategist_;
+        vaultManager = vaultManager_;
+        superformIds = superformIds_;
+        weights = startingWeights_;
+        depositLimit = depositLimit_;
     }
 
     //////////////////////////////////////////////////////////////
     //                  EXTERNAL  FUNCTIONS                     //
     //////////////////////////////////////////////////////////////
 
-    /// @notice Sets the deposit limit for the vault
-    /// @param depositLimit_ The new deposit limi
-    function setDepositLimit(uint256 depositLimit_) external onlySuperVaultsStrategist {
-        SV.depositLimit = depositLimit_;
+    /// @inheritdoc ISuperVault
+    function setDepositLimit(uint256 depositLimit_) external override onlyVaultManager {
+        depositLimit = depositLimit_;
 
         emit DepositLimitSet(depositLimit_);
     }
 
-    /// @notice Sets the refunds receiver address
-    /// @param refundReceiver_ The new refunds receiver address
-    function setRefundsReceiver(address refundReceiver_) external onlySuperVaultsStrategist {
-        if (refundReceiver_ == address(0)) revert ZERO_ADDRESS();
-        refundReceiver = refundReceiver_;
+    /// @inheritdoc ISuperVault
+    function setStrategist(address strategist_) external override onlyManagement {
+        strategist = strategist_;
+
+        emit StrategistSet(strategist_);
+    }
+
+    /// @inheritdoc ISuperVault
+    function setValidFormImplementationIds(uint32 ERC5115FormImplementationId_) external override onlyManagement {
```

Comment:
nit: this function sets ERC5115FormImplementationId but the function natspec and name are irrelevant

Reply:
fixed